lightningd/plugin.c

```diff
@@ -1086,9 +1087,35 @@ void json_add_opt_plugins(struct json_stream *response,
 			  const struct plugins *plugins)
 {
 	struct plugin *p;
+	struct plugin_opt *opt;
+	const char *opt_name;
+	const char *plugin_name;
+	char *plugin_ext;
+
+	/* we output 'plugins' and their config as substructure:
+	 * "plugins" : { "name" -> { "param" : "value" }, ... }  */
```

Comment:
i almost think it'd be nicer to put the name inside the object, so instead of a list of keyed objects we get a set of uniform objects. (this is easier on static parsers like in Go, i think). actually, it's probably even easier if we bucket the configs into their own dict then..

e.g.

```
plugins: [
    {
        "name": "hello",
        "path": "../hello/path../path",
        "options": {
            "option1": "value",
            "option2": "value"
        }
        },{
    ...
    }
]
```

Reply:
as an additional addendum, calling them 'options' is nice because it matches the language we use in the plugins documentation.

Reply:
ok. thanks for the feedback. done in latest force push